src/main/java/de/bonndan/nivio/output/jgraphx/RenderedXYMapFactory.java

```diff
@@ -0,0 +1,118 @@
+package de.bonndan.nivio.output.jgraphx;
+
+import com.mxgraph.model.mxCell;
+import com.mxgraph.model.mxGeometry;
+import com.mxgraph.view.mxGraph;
+import de.bonndan.nivio.model.LandscapeImpl;
+import de.bonndan.nivio.output.IconService;
+import de.bonndan.nivio.output.Rendered;
+import de.bonndan.nivio.output.map.*;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class RenderedXYMapFactory implements MapFactory<mxGraph, mxCell> {
+
+    public static final int SIZE_FACTOR = 40;
+    private final IconService iconService;
+
+    public RenderedXYMapFactory(IconService iconService) {
+        this.iconService = iconService;
+    }
+
+    public RenderedXYMap getRenderedMap(LandscapeImpl landscape, Rendered<mxGraph, mxCell> rendered) {
+        RenderedXYMap renderedMap = from(rendered);
+
+        AtomicInteger minX = new AtomicInteger(0);
+        AtomicInteger maxX = new AtomicInteger(0);
+        AtomicInteger minY = new AtomicInteger(0);
+        AtomicInteger maxY = new AtomicInteger(0);
+        renderedMap.items.forEach(item -> {
+            if (item.x < minX.get())
+                minX.set((int) item.x);
+            if (item.x > maxX.get())
+                maxX.set((int) item.x);
+            if (item.y < minY.get())
+                minY.set((int) item.y);
+            if (item.y > maxY.get())
+                maxY.set((int) item.y);
+        });
+
+        renderedMap.landscape = landscape.getName();
+        renderedMap.sizeFactor = SIZE_FACTOR;
+        renderedMap.width = maxX.get() - minX.get();
+        renderedMap.height = maxY.get() - minY.get();
+        int size = Math.max(renderedMap.width, renderedMap.height) / SIZE_FACTOR;
+
+        renderedMap.items.forEach(item -> item.size = size);
+        renderedMap.groups.forEach(groupMapItem -> groupMapItem.size = size);
+
+        return renderedMap;
+    }
+
+    RenderedXYMap from(Rendered<mxGraph, mxCell> rendered) {
+        RenderedXYMap renderedMap = new RenderedXYMap();
+
+        rendered.getItemObjects().forEach((item, cell) -> {
+            mxGeometry geometry = cell.getGeometry();
+
+            long x;
+            long y;
+            if (cell.getParent().getGeometry() != null) {
+                x = Math.round(geometry.getX() + cell.getParent().getGeometry().getX());
+                y = Math.round(geometry.getY() + cell.getParent().getGeometry().getY());
+            } else {
+                x = Math.round(geometry.getX());
+                y = Math.round(geometry.getY());
+            }
```

Comment:
Can't you extract this into a method (using some kind of Point)? It get's repeated in lines 93-102.

Reply:
done.